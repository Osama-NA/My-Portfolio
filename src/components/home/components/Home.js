import React from 'react'
import {HomeContainer} from '../styles/Home.styled.js';
import {ThreeDCubes} from './ThreeDCubes';
import {HelloButton} from './HelloButton';

export const Home = ({theme, cubesColor}) => {
    
    return (
        <HomeContainer data-testid="home">
            <div className="designs-container">
                <ThreeDCubes cubesColor={cubesColor}/>
                <HelloButton />
            </div>
            <main className="info-container">

            </main>
            <div className="quote">

            </div>
            <div className="github-icon">

            </div>
        </HomeContainer>
    )
}