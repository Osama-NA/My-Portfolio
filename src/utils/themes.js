// GREEN / GRAY
export const greenGrayTheme = {
    themeColors: {
        mainColor: '#54EBA0'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#03AD78',
    circleBackground: '#313138'
}

// BLUEISH PURPLE / GRAY 
export const blueishPurpleGrayTheme = {
    themeColors: {
        mainColor: '#674DFF'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#544DFF',
    circleBackground: '#313138'
}

// PURPLE / GRAY
export const purpleGrayTheme = {
    themeColors: {
        mainColor: '#B15EFF'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#9A52DE',
    circleBackground: '#313138'
}

// LIGHT BLUE / GRAY
export const lightBlueGrayTheme = {
    themeColors: {
        mainColor: '#00E6E2'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#00E6E2',
    circleBackground: '#313138'
}

// YELLOW / GRAY
export const yellowGrayTheme = {
    themeColors: {
        mainColor: '#F2F270'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#F2F208',
    circleBackground: '#313138'
}

// PINK / GRAY
export const pinkGrayTheme = {
    themeColors: {
        mainColor: '#FF558E'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#FF558E',
    circleBackground: '#313138'
}

// RED / GRAY
export const redGrayTheme = {
    themeColors: {
        mainColor: '#FF3142'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#FF3142',
    circleBackground: '#313138'
}