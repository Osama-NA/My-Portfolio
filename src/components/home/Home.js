import React from 'react'
import {HomeContainer} from './styles/Home.styled.js';
import {ThreeDCubes} from './components/ThreeDCubes';
import { HelloButton } from './components/HelloButton';
import { Info } from './components/Info';
import {Quote} from './components/Quote';
import { IconLinks } from './components/IconLinks';
import {ThemesButton} from '../ThemesButton/ThemesButton';

export const Home = ({cubesColor, circleBackground}) => {
    
    return (
        <HomeContainer data-testid="home">
            <div className="designs-container">
                <ThreeDCubes cubesColor={cubesColor}/>
                <HelloButton />
            </div>
            <Info circleBackground={circleBackground} />
            <footer>
                <Quote />
                <IconLinks />
                <ThemesButton />
            </footer>
        </HomeContainer>
    )
}