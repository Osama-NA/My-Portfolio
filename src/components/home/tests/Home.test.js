import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import { Home } from '../Home.js';
import {ThemeProvider} from 'styled-components';
import {BrowserRouter} from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';
import { lightPinkTheme } from '../../../utils/themes.js';

afterEach(cleanup);

describe("Home component tests", () => {

    const HomeComponent = () => {
        return (
            <BrowserRouter>
                <ThemeProvider theme={lightPinkTheme}>
                    <Home 
                        theme={lightPinkTheme}
                        cubesColor="lightpink" 
                        circleBackground="lightblue"
                    />
                </ThemeProvider>
            </BrowserRouter>
        )
    }

    it("Home component renders without crashing", () => {
        let div = document.createElement('div');
        ReactDOM.render(<HomeComponent />, div);
    })

    it("Home component is in the document", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toBeInTheDocument();
    })

    it("Home component contains HelloButton component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('hello-button'));
    })

    it("Home component contains ThreeDCubes component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('threed-cubes'));
    })

    it("Home component contains Info component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('info'));
    })

    it("Home component contains Quote component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('quote'));
    })

    it("Home component contains IconLinks component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('icon-links'));
    })

    it("Home component matches snapshot", () => {
        const tree = renderer.create(<HomeComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})