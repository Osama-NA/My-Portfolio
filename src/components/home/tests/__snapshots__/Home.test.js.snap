// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests Home component matches snapshot 1`] = `
<div
  className="sc-bdfBQB epEUgK"
  data-testid="home"
>
  <div
    className="designs-container"
  >
    <div
      className="sc-gsTEea KETHo"
      data-testid="threed-cubes"
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <canvas
          style={
            Object {
              "display": "block",
            }
          }
        />
      </div>
    </div>
    <div
      className="sc-dlfnuX douitY"
      data-testid="hello-button"
    >
      <p
        className="hello-btn"
        onClick={[Function]}
      >
        Hello there!
      </p>
      <div
        className="message-box"
        data-testid="message-box"
        style={Object {}}
      >
        <p>
          Donâ€™t waste your time clicking a hello text,
          <br />
          <a
            data-testid="message-box-link"
            href="/Router/Projects"
            onClick={[Function]}
          >
            view my work here.
          </a>
        </p>
        <i
          className="fas fa-times close-btn"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <main
    className="sc-hKgJUU PZkbk"
    data-testid="info"
  >
    <h1>
      Osama Al Hajj Ali
    </h1>
    <h3>
      I am a web developer
    </h3>
    <nav>
      <ul
        data-testid="nav-bar"
      >
        <li>
          <a
            data-testid="projects-link"
            href="/Router/Projects"
            onClick={[Function]}
          >
            Projects
          </a>
        </li>
        <li>
          <a
            data-testid="contacts-link"
            href="/Router/Contacts"
            onClick={[Function]}
          >
            Contact me
          </a>
        </li>
        <li>
          <a
            data-testid="about-link"
            href="/Router/About"
            onClick={[Function]}
          >
            About me
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="globe-container"
    >
      <div
        className="globe"
        data-testid="globe"
      >
        <div
          style={
            Object {
              "height": "100%",
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <canvas
            style={
              Object {
                "display": "block",
              }
            }
          />
        </div>
      </div>
    </div>
  </main>
  <footer>
    <div
      className="sc-eCstlR bYgCvX"
    >
      <i
        className="fas fa-quote-left quote"
      />
      <p
        className="learn"
      >
        
        _
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          .
        </span>
      </p>
      <p
        className="develop"
      >
        
        
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          .
        </span>
      </p>
      <p
        className="mac"
      >
        
        
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          .
        </span>
      </p>
    </div>
    <div
      className="sc-jSgvzq eGTnut"
    >
      <div
        className="circle"
      />
      <a
        href="https://github.com/Osama-NA"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-github"
        />
      </a>
      <div
        className="circle"
      />
    </div>
  </footer>
</div>
`;
